### PR TITLE
add data_scanned to Athena

### PR DESCRIPTION
**Pull request by alison985**
Athena costs are based on how much data is scanned by the query. This PR leverages the PyAthena library functionality to report that information to the user. It also saves it to the database so it can be analyzed later.

### CODE REVIEW

File: redash/models.py [hunk 1]
```diff
@@ -617,18 +617,25 @@ class QueryResult(db.Model, BelongsToOrgMixin):
...
     retrieved_at = Column(db.DateTime(True))
+    data_scanned = Column(db.String(255), nullable=True)
 
     __tablename__ = 'query_results'
 
     def to_dict(self):
+        if hasattr(self, 'data_scanned'):
+            data_scanned_info = self.data_scanned
+        else:
+            data_scanned_info = 'to_dict'
```
Comment (arikfr) [inline redash/models.py]: Why `to_dict`? 

Btw, this `if` check will be always true...

File: redash/models.py [hunk 2]
```diff
@@ -663,13 +670,20 @@ def get_latest(cls, data_source, query, max_age=0):
 
     @classmethod
     def store_result(cls, org, data_source, query_hash, query, data, run_time, retrieved_at):
+        try:
+            data_scanned_information = json.loads(data)['data_scanned']
+        except (ValueError, TypeError) as e:
```
Comment (arikfr) [inline redash/models.py]: You can use `json.loads(data).get('data_scanned', 'error')` which will return `None` if it has no value. No need for the `try`/`except`.

But why make every other query runner return some value here when this is probably relevant for 2-3 data sources at the moment? 

I think a simpler approach will be to return nothing when it's not supported, and just not show any value in the UI.

---

Another issue is that calling `json.loads` here can be a performance/memory issue for large datasets. I have a different approach to suggest, but will outline it in the review summary.

File: redash/query_runner/athena.py [hunk 3]
```diff
@@ -149,7 +149,13 @@ def run_query(self, query, user):
             column_tuples = [(i[0], _TYPE_MAPPINGS.get(i[1], None)) for i in cursor.description]
             columns = self.fetch_columns(column_tuples)
             rows = [dict(zip(([c['name'] for c in columns]), r)) for i, r in enumerate(cursor.fetchall())]
-            data = {'columns': columns, 'rows': rows}
+            qbytes = 'N/A'
+            try:
+                cursor._poll()
+                qbytes = cursor._data_scanned_in_bytes
+            except AttributeError as e:
+                debug("Athena Upstream can't get data_scanned_in_bytes: %s", e)
```
Comment (arikfr) [inline redash/query_runner/athena.py]: What's `debug`?

**Comment by rohanpd** [comment]
I'd love to see something like this implemented. At the moment we see users scanning lots of data from Athena hiking up our bills without them even knowing it.

**Comment by arikfr** [comment]
Replaced by #2204.

**Pull request closed without merging**